packages/server/src/api/controllers/row/external.ts

```diff
@@ -102,6 +103,11 @@ export async function patch(ctx: UserCtx<PatchRowRequest, PatchRowResponse>) {
 
 export async function destroy(ctx: UserCtx) {
   const source = await utils.getSource(ctx)
+
+  if (sdk.views.isView(source) && helpers.views.isCalculationView(source)) {
```

Comment:
Can this not be done at the higher level, so it is not duplicated between external and internal?

Reply:
It can, though it would cause a refetch of the table/view. I put them where I did because it didn't cause any extra CouchDB fetches. It might be possible to have the best of both worlds but it would be a larger change.

Reply:
After discussing it, let's leave it as it is. This is the common pattern now, we should check if we can improve it in the future clean up, rethinking how we do differentiate between internal and external logic